cmd/pinniped-proxy/src/main.rs

```diff
@@ -28,34 +32,58 @@ async fn main() -> Result<()> {
     let default_ca_data = fs::read_to_string(opt.default_ca_cert.clone())
         .with_context(|| format!("unable to load default-ca-cert at {}", opt.default_ca_cert))?;
 
-    // For every incoming connection, we make a new hyper `Service` to handle
-    // all incoming HTTP requests on that connection. This is done by passing a
-    // closure to the hyper `make_service_fn` which returns our custom `make_svc`
-    // function that can be used for each connection.
-    let make_svc = make_service_fn(|_conn| {
-        let default_ca_data = default_ca_data.clone();
-        // The closure just returns an async block that runs a service to handle
-        // all requests for the connection.
-        async {
-            // `service_fn` is a helper from the hyper crate which converts a
-            // function that returns a Response into a `Service`. We pass a
-            // closure to service_fn here so we can pass the default certificate
-            // authority data.
-            Ok::<_, Infallible>(service_fn(move |req| {
-                service::proxy(req, default_ca_data.clone().into_bytes())
-            }))
-        }
-    });
-
     let addr = ([0, 0, 0, 0], opt.port).into();
 
-    let server = Server::bind(&addr).serve(make_svc);
-
-    info!("Listening on http://{}", addr);
+    let with_tls = opt.proxy_tls_cert != "" && opt.proxy_tls_cert_key != "";
```

Comment:
Maybe worth adding a warning if just one of them is set ?

Reply:
Even erroring and stopping - we can't run with just one.